### PR TITLE
Add AppSettings update as part of collector upgrade.

### PR DESCRIPTION
**Pull request by imranalisyed506**
### Problem Description
[ehub] SPIKE: check runtime version update to 4
### Solution Description
updated function runtime version from 3 to 4

### CODE REVIEW

File: updater.js [hunk 1]
```diff
@@ -28,15 +30,21 @@ const WebSiteManagement = require('@azure/arm-appservice');
...
-    constructor({clientId, domain, clientSecret, subscriptionId, resourceGroup, webAppName} = {}) {
+    constructor({ clientId, domain, clientSecret, subscriptionId, resourceGroup, webAppName } = {}) {
         this._clientId = clientId ? clientId : process.env.CUSTOMCONNSTR_APP_CLIENT_ID;
         this._domain = domain ? domain : process.env.APP_TENANT_ID;
         this._clientSecret = clientSecret ? clientSecret : process.env.CUSTOMCONNSTR_APP_CLIENT_SECRET;
         this._subscriptionId = subscriptionId ? subscriptionId : process.env.APP_SUBSCRIPTION_ID;
         this._resourceGroup = resourceGroup ? resourceGroup : process.env.APP_RESOURCE_GROUP;
         this._webAppName = webAppName ? webAppName : process.env.WEBSITE_SITE_NAME;
+        this._azureWebsiteClient = new WebSiteManagementClient(this._getAzureCredentials(), this._subscriptionId);
+        this.azureClientObject = {
```
Comment (kkuzmin) [inline updater.js]: I suggest we rename it to `azureWebsiteClientObject `

File: updater.js [hunk 2]
```diff
@@ -48,12 +56,39 @@ class AlAzureUpdater {
...
+        var updateEnv = envObject;
+        m_util.updateAppSettings(updateEnv, this.azureClientObject, function (settingsError) {
+            if (settingsError) {
+                return callback(settingsError);
+            } else {
+                return callback(null);
+            }
+        });
+    }
+    run(envObject, callback) {
```
Comment (kkuzmin) [inline updater.js]: I suggest we don't pass `envObject` as parameter. But rather have updater reading it itself.
```
const envObject = fs:readFile(process.env.azure_fun_update_config_name);
```

And have `AZURE_FUN_UPDATE_CONFIG_NAME = ../al-ehub-collector.json` env var set in the [template](https://github.com/alertlogic/ehub-collector/blob/v1/templates/ehub.json#L121-L158)

**Review by kkuzmin — CHANGES_REQUESTED** [review]

File: updater.js (hunk 1 above)
Comment (imranalisyed506) [reply updater.js]: resolved

File: updater.js [hunk 3]
```diff
@@ -48,12 +56,57 @@ class AlAzureUpdater {
...
+            }
+        });
+    }
+    
+    readEnvFile(callback){
+        fs.readFile(process.cwd() + "/"+ process.env.AZURE_FUN_UPDATE_CONFIG_NAME,'utf-8',function (err, data) {
+          if (err) {
+             return callback(err);
+            } else{
+              const envData = JSON.parse(data);
```
Comment (kkuzmin) [inline updater.js]: please wrap `JSON.parse` into `try..catch` and return `callback(exception)`

**Review by kkuzmin — CHANGES_REQUESTED** [review]

File: updater.js [hunk 4]
```diff
@@ -48,12 +56,57 @@ class AlAzureUpdater {
...
+            if (settingsError) {
+                return callback(settingsError);
+            } else {
+                return callback(null);
+            }
+        });
+    }
+    
+    readEnvFile(callback){
+        fs.readFile(process.cwd() + "/"+ process.env.AZURE_FUN_UPDATE_CONFIG_NAME,'utf-8',function (err, data) {
```
Comment (kkuzmin) [inline updater.js]: please add spaces between parameters

File: updater.js (hunk 3 above)
Comment (imranalisyed506) [reply updater.js]: resolved

File: updater.js (hunk 4 above)
Comment (imranalisyed506) [reply updater.js]: > please add spaces between parameters

resolved

File: util.js [hunk 5]
```diff
@@ -52,8 +52,51 @@ const verifyObjProps = function (obj, expectedProps) {
...
+    }
+
+
 
 module.exports = {
         getADCacheFilename: getADCacheFilename,
-        verifyObjProps: verifyObjProps
+        verifyObjProps: verifyObjProps,
+        updateAppSettings: updateAppSettings
+
```
Comment (kkuzmin) [inline util.js:103]: let's export `getAppSettings()` and `setAppSettings()` just in case

**Review by kkuzmin — APPROVED** [review]

**Pull request merged**